### PR TITLE
Revert onCreate changes

### PR DESCRIPTION
**Pull request by WhatsEmo**
## Summary

Revert ReaderSdk.initialize changes in TatsuUkraine:flutter-3. This change causes the application to crash whenever the app is background and the app has been initialized before. This is a fix for an on-going issue a developer is facing.

## Related issues

Fix #

## Changelog

- Move ReaderSdk.initialize back to Application's onCreate
- Delete ReaderSdk.initialize related code in the Plugin's Activity (`SquareReaderSdkFlutterPlugin.java`)

## Test Plan

Manual tests will be preformed to test these changes

### CODE REVIEW

File: example/android/app/src/main/java/com/example/flutter/squareup/sdk/reader/MainActivity.kt [hunk 1]
```diff
@@ -15,8 +15,17 @@ limitations under the License.
...
 
-class MainActivity: FlutterActivity() {
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.plugins.GeneratedPluginRegistrant;
+
+public class MainActivity extends FlutterActivity {
+
+    @Override
+    public void configureFlutterEngine(@NonNull FlutterEngine flutterEngine) {
```
[the remark this answers is not in the document]
Comment (plinio-square) [reply example/android/app/src/main/java/com/example/flutter/squareup/sdk/reader/MainActivity.kt:27]: I think (and this is stretching my knowledge on Android) that plugins don't necessarily have access to the main activity, so configureFlutterEngine and registering the plugin so the methods are available on the activity with `registerWith` is what this piece of code is doing. However, I'm looking at the docs and I see that on Flutter 2+ we should do this differently? https://docs.flutter.dev/release/breaking-changes/plugin-api-migration 

Per that page:
```
public class MainActivity extends FlutterActivity {
   // You can keep this empty class or remove it. Plugins on the new embedding
   // now automatically registers plugins.
 }
```

File: example/android/app/src/main/java/com/example/flutter/squareup/sdk/reader/MainApplication.kt [hunk 2]
```diff
@@ -15,7 +15,22 @@ limitations under the License.
 */
 package com.example.flutter.squareup.sdk.reader;
 
-import io.flutter.app.FlutterApplication
+import android.app.Activity;
 
-class MainApplication: FlutterApplication() {
+import androidx.annotation.CallSuper;
```
[the remark this answers is not in the document]
Comment (WhatsEmo) [reply example/android/app/src/main/java/com/example/flutter/squareup/sdk/reader/MainApplication.kt:20]: I'll remove this!

**Review by fka3 — APPROVED** [review]
At least yesterday, CI was red... but as I read this code, it looks good.  I'd nag @Nilay-squareup to see if he approves or knows some Flutter thing we're missing.  But if we don't hear in another 24hrs (i.e. another Indian day shift) and the CI is green, I'm good with this.

**Comment by WhatsEmo** [comment]
@Nilay-squareup Flutter is sort of outside my domain, so the failing CI is bizarre to me since I didn't change files related to those library files. I looked around and the suggested fix is to bump the Flutter version?

**Comment by Nilay-squareup** [comment]
> @Nilay-squareup Flutter is sort of outside my domain, so the failing CI is bizarre to me since I didn't change files related to those library files. I looked around and the suggested fix is to bump the Flutter version?

I think

**Comment by Nilay-squareup** [comment]
@WhatsEmo The CI is failing because some of the properties used in styles have been deprecated in recent version of flutter, And I already have fixed this issue in My [PR](https://github.com/square/reader-sdk-flutter-plugin/pull/114) which is not yet merged. and regarding the code I have seen that you have placed java code in `.kt` files, So we should convert the code to kotlin.

cc: @fka3

**Pull request closed without merging**